javascript/ql/lib/semmle/javascript/PrintAst.qll

```diff
@@ -500,9 +505,11 @@ private module PrintJavaScript {
     override Parameter element;
 
     override AstNode getChildNode(int childIndex) {
-      childIndex = 0 and result = element.getTypeAnnotation()
+      result = super.getChildNode(childIndex) // in case the parameter is a destructuring pattern
```

Comment:
Nice trick with the negative indices, I trust the destructuring does not use those?

Reply:
Yep, that starts at 0 (like all normal indices).

Reply:
Then I see no issue with this.